packages/jupyter-ai/jupyter_ai/chat_handlers/ask.py

```diff
@@ -49,11 +49,7 @@ def __init__(self, *args, **kwargs):
 
         self.parser.prog = "/ask"
         self.parser.add_argument("query", nargs=argparse.REMAINDER)
-        learn_chat_handler = self.chat_handlers.get("/learn")
-        if not isinstance(learn_chat_handler, LearnChatHandler):
-            raise CustomLearnException()
-
-        self._retriever = Retriever(learn_chat_handler=learn_chat_handler)
+        self._retriever = None  # Initialize lazily
```

Comment:
If you accept the above suggestion, it would also be good to update this comment to warn developers against accessing this directly:

```suggestion
        # NOTE: devs should use self.retriever property to access the retriever
        self._retriever = None
```